Shared/Extensions/URLExtensions.swift

```diff
@@ -468,6 +468,39 @@ extension URL {
     }
 }
 
+extension URL {
+    public var isBookmarklet: Bool {
+        return self.absoluteString.isBookmarklet
+    }
+    
+    public var bookmarkletCodeComponent: String? {
+        return self.absoluteString.bookmarkletCodeComponent
+    }
+}
+
+extension String {
+    public var isBookmarklet: Bool {
+        let url = self.lowercased()
+        return url.hasPrefix("javascript:") && !url.hasPrefix("javascript:/")
+    }
+    
+    public var bookmarkletCodeComponent: String? {
```

Comment:
Can you add a simple unit test here? One success assertion and one nil assertion